source/Octopus.Tentacle.Tests.Integration/FileTransferServiceTests.cs

```diff
@@ -16,22 +16,30 @@ public class FileTransferServiceTests : IntegrationTest
     {
         [Test]
         [TestCaseSource(typeof(TentacleTypesToTest))]
-        public async Task UploadFileSuccessfully(TentacleType tentacleType)
+        public async Task UploadFileSuccessfully(TentacleType tentacleType, SyncOrAsyncHalibut syncOrAsyncHalibut)
         {
             using var fileToUpload = new RandomTemporaryFileBuilder().Build();
 
             using var clientAndTentacle = await new LegacyClientAndTentacleBuilder(tentacleType)
+                .WithAsyncHalibutFeature(syncOrAsyncHalibut.ToAsyncHalibutFeature())
                 .Build(CancellationToken);
 
-            var uploadResult = clientAndTentacle.TentacleClient.FileTransferService.UploadFile(
-                "the_remote_uploaded_file",
-                new DataStream(
-                    fileToUpload.File.Length,
-                    stream =>
-                    {
-                        using var fileStream = File.OpenRead(fileToUpload.File.FullName);
-                        fileStream.CopyTo(stream);
-                    }));
+#pragma warning disable CS0612
+            var dataStream = new DataStream(
+                fileToUpload.File.Length,
+                stream =>
+                {
+                    using var fileStream = File.OpenRead(fileToUpload.File.FullName);
```

Comment:
Why not pass a async implementation as well?

Reply:
I haven't fixed these up yet - see commits - it's got added this morning as this made it into main yessturday